### PR TITLE
Add a simpler SoundCloud search command

### PR DESCRIPTION
**Pull request by zeyla**
The current method of searching via soundcloud is by prefixing the
search query for the play command with "scsearch:". In total, this is
used like `!!!play scsearch:rameses b mountains`. This is non-ergonomic
for the user.

Instead, create a second command specifically for this platform, used
like `!!!soundcloud rameses b mountains`, with `!!!sc` as a shortcut.

To make this more feasible without code duplication, the core logic has
been moved out to an abstract `BasePlayCommand`, leaving query
transformation logic (prefixing `scsearch` to the query) to be
overridden, and a method to be implemented with the purpose of
customizing the message sent when no arguments are provided.

tl;dr:

Old command usage:
`!!!play scsearch:rameses b mountains`

New command usage:
`!!!soundcloud rameses b mountains` or `!!!sc rameses b mountains`

### CODE REVIEW

**Comment by bramhaag** [comment]
Not using `@Override` makes me sad
